packages/app/src/components/swap/swap.tsx

```diff
@@ -64,7 +64,8 @@ const Swap: FC<Props> = ({
   const { account } = useEthers();
 
   const allowance = useTokenAllowance(tokenSendedAddress, account, addresses[protocolName].swap);
-  const isTokenApproved = allowance != null && allowance.gte(sendTokenAmount === "" ? "0" : sendTokenAmount);
+  const isTokenApproved =
+    allowance != null && allowance.gte(sendTokenAmount === "" ? "0" : utils.parseEther(sendTokenAmount));
```

Comment:
Since `""` is a falsy value this can be shorted to `allowance?.gte(sendTokenAmount || "0")`

- If allowance is undefined this will eval as `false` 
-  If sendTokenAmount is "" this will eval as `false`

Reply:
it is falsy, but it's not false, so if we want to use it as an input later, we have to change the prop types of the component, or do the conversion